Socket authentication failing due to missing module and security false positives

## Problem
Users cannot sign in successfully due to multiple socket authentication issues that are blocking real-time functionality.

## Error Analysis from Logs

### 1. 🚨 **Missing Socket Authentication Module**
```
❌ Socket authentication error: [Error: Cannot find module '/app/lib/socket-auth-secure.mjs' imported from /app/server.js]
🚫 Failed to authenticate socket, disconnecting: Gom6127dExmutJALAAAB
```

**Root Cause**: The socket authentication module `/app/lib/socket-auth-secure.mjs` is missing, causing all socket connections to fail.

### 2. 🛡️ **Security System False Positives**
```
🚨 SECURITY THREAT DETECTED: {
  securityResult: {
    isValid: false,
    shouldRotateSession: false,
    shouldClearCookies: true,
    reason: 'Insecure transmission in production environment'
  },
  hijackingCheck: {
    isHijackingAttempt: true,
    riskLevel: 'high',
    indicators: [ 'Suspicious header detected: x-forwarded-host' ]
  }
}
```

**Issues**:
- **False HTTPS Detection**: System incorrectly identifies localhost as "insecure transmission in production"
- **Proxy Header Misinterpretation**: `x-forwarded-host` header (common in reverse proxies) flagged as hijacking attempt
- **Overly Aggressive Security**: Legitimate local development/staging requests blocked

### 3. 🔄 **Authentication Flow Breakdown**
- User authentication succeeds initially (token found and validated)
- Security validation triggers false positives
- Socket authentication fails due to missing module
- User effectively locked out despite valid credentials

## Impact
- **Critical**: Users cannot sign in to the application
- **Real-time Features Broken**: Socket connections fail immediately
- **Development/Staging Blocked**: Local and staging environments unusable
- **False Security Alerts**: Legitimate traffic flagged as threats

## Technical Requirements

### Immediate Fixes Needed
1. **Create/Fix Socket Authentication Module**
   - Ensure `/app/lib/socket-auth-secure.mjs` exists and is properly deployed
   - Verify module exports expected authentication functions
   - Test socket connection authentication flow

2. **Fix Security Detection Logic**
   - Update HTTPS detection to handle localhost and staging environments
   - Whitelist legitimate proxy headers (`x-forwarded-host`, `x-forwarded-proto`)
   - Adjust security thresholds for development/staging environments
   - Add environment-aware security rules

3. **Environment-Specific Configuration**
   - **Development**: Relaxed security for localhost
   - **Staging**: Allow proxy headers but maintain security
   - **Production**: Full security enforcement with proper HTTPS detection

## Acceptance Criteria
- [ ] Socket authentication module exists and loads successfully
- [ ] Users can sign in and maintain socket connections
- [ ] Security system allows legitimate localhost/staging requests
- [ ] Proxy headers properly handled in staging environments
- [ ] Real-time features work after successful authentication
- [ ] No false positive security alerts for legitimate traffic
- [ ] Environment-specific security configurations implemented

## Environment Context
- **URL**: http://localhost:3000/dashboard
- **User Agent**: Chrome 138.0.0.0 on macOS
- **IP**: 10.0.1.1 (internal network)
- **Headers**: x-forwarded-host present (legitimate proxy setup)

## Security Notes
While fixing these issues, maintain appropriate security measures for production while allowing legitimate development and staging workflows.

## Related Files
- `server.js` - Socket server configuration
- `lib/socket-auth-secure.mjs` - Missing authentication module
- Security validation logic (needs identification)
- Environment configuration files

--- a/retro-ai/Dockerfile
+++ b/retro-ai/Dockerfile
@@ -74,6 +74,9 @@ COPY --from=builder --chown=nextjs:nodejs /app/.next ./.next
 # Copy custom server
 COPY --from=builder --chown=nextjs:nodejs /app/server.js ./server.js
 
+# Copy lib directory with socket authentication modules
+COPY --from=builder --chown=nextjs:nodejs /app/lib ./lib
+
 # Copy production dependencies from separate stage
 COPY --from=prod-deps --chown=nextjs:nodejs /app/node_modules ./node_modules
 

--- a/retro-ai/__tests__/cookie-security.test.tsx
+++ b/retro-ai/__tests__/cookie-security.test.tsx
@@ -142,6 +142,40 @@ describe('Cookie Security Utilities', () => {
       
       process.env.NODE_ENV = originalEnv;
     });
+
+    it('should accept HTTPS via proxy headers in production', async () => {
+      const originalEnv = process.env.NODE_ENV;
+      process.env.NODE_ENV = 'production';
+      
+      mockGetToken.mockResolvedValue({
+        iat: Math.floor(Date.now() / 1000),
+        exp: Math.floor(Date.now() / 1000) + 3600
+      });
+      
+      const mockRequest = {
+        headers: new Map([
+          ['user-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'],
+          ['x-forwarded-proto', 'https']
+        ]),
+        cookies: new Map(),
+        method: 'GET',
+        url: 'http://staging-retroai.tryitnow.dev/dashboard' // HTTP URL but HTTPS via proxy
+      } as unknown as NextRequest;
+
+      mockRequest.headers.get = jest.fn((key: string) => {
+        const headers: Record<string, string> = {
+          'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)',
+          'x-forwarded-proto': 'https'
+        };
+        return headers[key] || null;
+      });
+
+      const result = await validateCookieSecurity(mockRequest);
+      
+      expect(result.isValid).toBe(true);
+      
+      process.env.NODE_ENV = originalEnv;
+    });
   });
 
   describe('createSecureCookieHeaders', () => {
@@ -254,7 +288,7 @@ describe('Cookie Security Utilities', () => {
       const mockRequest = {
         headers: new Map([
           ['user-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'],
-          ['x-forwarded-host', 'malicious.com']
+          ['x-original-url', 'malicious.com']
         ]),
         cookies: new Map()
       } as unknown as NextRequest;
@@ -265,7 +299,27 @@ describe('Cookie Security Utilities', () => {
 
       expect(result.isHijackingAttempt).toBe(true);
       expect(result.riskLevel).toBe('high');
-      expect(result.indicators).toContain('Suspicious header detected: x-forwarded-host');
+      expect(result.indicators).toContain('Suspicious header detected: x-original-url');
+    });
+
+    it('should allow x-forwarded-host header (not suspicious)', () => {
+      const mockRequest = {
+        headers: new Map([
+          ['user-agent', 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'],
+          ['x-forwarded-host', 'staging-retroai.tryitnow.dev']
+        ]),
+        cookies: new Map()
+      } as unknown as NextRequest;
+
+      mockRequest.cookies.getAll = jest.fn().mockReturnValue([
+        { name: 'next-auth.session-token', value: 'valid-token' }
+      ]);
+
+      const result = detectSessionHijacking(mockRequest);
+
+      expect(result.isHijackingAttempt).toBe(false);
+      expect(result.riskLevel).toBe('low');
+      expect(result.indicators).toHaveLength(0);
     });
 
     it('should detect suspicious user agent', () => {

--- a/retro-ai/lib/cookie-security.ts
+++ b/retro-ai/lib/cookie-security.ts
@@ -163,7 +163,12 @@ export async function validateCookieSecurity(
 
     // Check for secure transmission
     if (process.env.NODE_ENV === 'production') {
-      if (!req.url.startsWith('https://')) {
+      // Check if request is over HTTPS (including behind proxies)
+      const isSecure = req.url.startsWith('https://') || 
+                      req.headers.get('x-forwarded-proto') === 'https' ||
+                      req.headers.get('x-forwarded-ssl') === 'on';
+      
+      if (!isSecure) {
         return {
           isValid: false,
           shouldRotateSession: false,
@@ -305,8 +310,12 @@ export function detectSessionHijacking(req: NextRequest): {
   // Check for suspicious headers (with environment-aware detection)
   const suspiciousHeaders = ['x-original-url', 'x-rewrite-url'];
   
-  // Only treat x-forwarded-host as suspicious in production environments
-  if (!isDevelopment) {
+  // Only treat x-forwarded-host as suspicious in true production (not staging)
+  // Staging environments commonly use reverse proxies with this header
+  const isProduction = process.env.NODE_ENV === 'production';
+  const isStaging = req.url.includes('staging') || req.headers.get('host')?.includes('staging');
+  
+  if (isProduction && !isStaging) {
     suspiciousHeaders.push('x-forwarded-host');
   }
   

--- a/retro-ai/lib/cookie-security.ts
+++ b/retro-ai/lib/cookie-security.ts
@@ -310,14 +310,8 @@ export function detectSessionHijacking(req: NextRequest): {
   // Check for suspicious headers (with environment-aware detection)
   const suspiciousHeaders = ['x-original-url', 'x-rewrite-url'];
   
-  // Only treat x-forwarded-host as suspicious in true production (not staging)
-  // Staging environments commonly use reverse proxies with this header
-  const isProduction = process.env.NODE_ENV === 'production';
-  const isStaging = req.url.includes('staging') || req.headers.get('host')?.includes('staging');
-  
-  if (isProduction && !isStaging) {
-    suspiciousHeaders.push('x-forwarded-host');
-  }
+  // Note: x-forwarded-host header is allowed in all environments
+  // as reverse proxies commonly use this header for legitimate routing
   
   for (const header of suspiciousHeaders) {
     if (req.headers.get(header)) {

